src/pages/[platform]/build-a-backend/add-aws-services/analytics/storing-data/index.mdx

```diff
@@ -0,0 +1,204 @@
+
+// Define the S3 bucket resource
+export const storage = defineStorage({
+  name: "FirehoseDestinationBucket",
+});
+
+```
+
+next, let's create the Amazon Data Firehose resource.
+
+```ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import { storage } from "./storage/resource";
+import { CfnDeliveryStream } from "aws-cdk-lib/aws-kinesisfirehose";
+import { Stack } from "aws-cdk-lib/core";
+import {
+  Policy,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from "aws-cdk-lib/aws-iam";
+
+const backend = defineBackend({
+  auth, 
+  data,
+  storage,
+  // additional resources 
+});
+
+// Create a new stack for the Firehose resources
+const firehoseStack = backend.createStack("firehose-stack");
+
+// Access the S3 bucket resource
+const s3Bucket = backend.storage.resources.bucket;
+
+// Create a new IAM role for the Firehose
+const firehoseRole = new Role(firehoseStack, "FirehoseRole", {
+  assumedBy: new ServicePrincipal("firehose.amazonaws.com"),
+});
+
+// Grant the Firehose role read/write permissions to the S3 bucket
+s3Bucket.grantReadWrite(firehoseRole);
+
+// Create a new Firehose delivery stream
+const myFirehose = new CfnDeliveryStream(firehoseStack, "MyFirehose", {
+  deliveryStreamType: "DirectPut",
+  s3DestinationConfiguration: {
+    bucketArn: s3Bucket.bucketArn,
+    roleArn: firehoseRole.roleArn,
+  },
+  deliveryStreamName: "myFirehose",
+});
+
+// Create a new IAM policy to allow users to write to the Firehose
+const firehosePolicy = new Policy(firehoseStack, "FirehosePolicy", {
+  statements: [
+    new PolicyStatement({
+      actions: ["firehose:PutRecordBatch"],
+      resources: [myFirehose.attrArn],
+    }),
+  ],
+});
+
+// Attach the policy to the authenticated and unauthenticated IAM roles
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(firehosePolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(firehosePolicy);
+
+// Patch the custom Analytics resource to the output configuration
+backend.addOutput({
+  custom: {
+    KinesisFirehose: {
+      firehoseName: myFirehose.ref,
+      region: Stack.of(firehoseStack).region,
+    },
+  },
+});
+
+```
+
+## Installation and Configuration
+
+Ensure you have [setup IAM permissions](https://docs.aws.amazon.com/firehose/latest/dev/controlling-access.html) for `firehose:PutRecordBatch`.
+
+Example IAM policy for Amazon Data Firehose:
+
+```javascript
+{
+  "Version": "2012-10-17",
+  "Statement": [{
+    "Effect": "Allow",
+    "Action": "firehose:PutRecordBatch",
+    // replace the template fields
+    "Resource": "arn:aws:firehose:<your-aws-region>:<your-aws-account-id>:deliverystream/<your-stream-name>"
+  }]
+}
+```
+
+Configure Kinesis Firehose:
+
+```javascript
+import { Amplify } from 'aws-amplify';
+
+Amplify.configure({
+  ...Amplify.getConfig(),
+  Analytics: {
+    KinesisFirehose: {
```

Comment:
are the custom outputs needed for this config?

Reply:
not really, but was intended to show an example.
updated the example